src/hash.cr

```diff
@@ -1010,6 +1017,43 @@ class Hash(K, V)
     updated_entry ? updated_entry.value : yield key
   end
 
+  # Updates the current value of *key* with the value returned by the given block
+  # (the current value is used as input for the block).
+  #
+  # If no entry for *key* is present, but there's a default value (or default block)
+  # then that default value is used as input for the given block.
+  #
+  # If no entry for *key* is present and the hash has no default value, it raises `KeyError`.
+  #
+  # It returns the value used as input for the given block
+  # (ie. the old value if key present, or the default value)
+  #
+  # ```
+  # h = {"a" => 0, "b" => 1}
+  # h.update("b") { |v| v + 41 } # => 1
+  # h["b"]                       # => 42
+  #
+  # h = Hash(String, Int32).new(40)
+  # h.update("foo") { |v| v + 2 } # => 40
+  # h["foo"]                      # => 42
+  #
+  # h = {} of String => Int32
+  # h.update("a") { 42 } # raises KeyError
+  # ```
```

Comment:
Maybe a reference to `#transform_values!` which has similar mechanics might be useful?

```suggestion
  # ```
  #
  # See `#transform_values!` for updating *all* the values.
```

Similar a back reference from that methods doc.